chanbackup/backupfile_test.go

```diff
@@ -45,160 +45,217 @@ func assertFileDeleted(t *testing.T, filePath string) {
+			// the directory between test cases), we verify the
+			// directory doesn't exist at all.
+			if tc.noBackupArchive {
+				require.NoDirExists(t, archiveDir)
+				return
+			}
+
+			// Otherwise we expect an archive to be created.
+			files, err := os.ReadDir(archiveDir)
+			require.NoError(t, err)
+			require.Len(t, files, 1)
+
+			// Verify the archive contents match the previous
+			// backup.
+			archiveFile := filepath.Join(
+				archiveDir, files[0].Name(),
+			)
+
+			// The archived content should match the previous backup
+			// (newPackedMulti) that was just swapped out.
+			assertBackupMatches(t, archiveFile, newPackedMulti)
+
+			// Clean up the archive directory.
+			os.RemoveAll(archiveDir)
+		})
+	}
+}
+
 // TestUpdateAndSwap test that we're able to properly swap out old backups on
 // disk with new ones. Additionally, after a swap operation succeeds, then each
 // time we should only have the main backup file on disk, as the temporary file
-// has been removed. Finally, we check for noBackupArchive to ensure that the
-// archive file is created when it's set to false, and not created when it's
-// set to true.
+// has been removed.
 func TestUpdateAndSwap(t *testing.T) {
 	t.Parallel()
 
-	tempTestDir := t.TempDir()
-
-	testCases := []struct {
-		fileName     string
-		tempFileName string
+	// Check that when the main file name is blank, an error is returned.
+	backupFile := NewMultiFile("", false)
 
-		oldTempExists   bool
-		noBackupArchive bool
+	err := backupFile.UpdateAndSwap(PackedMulti(nil))
+	require.ErrorIs(t, err, ErrNoBackupFileExists)
 
-		valid bool
+	testCases := []struct {
+		name          string
+		oldTempExists bool
 	}{
-		// Main file name is blank, should fail.
-		{
-			fileName: "",
-			valid:    false,
-		},
-
 		// Old temporary file still exists, should be removed. Only one
 		// file should remain.
 		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
+			name:          "remove old temp file",
 			oldTempExists: true,
-			valid:         true,
 		},
 
 		// Old temp doesn't exist, should swap out file, only a single
 		// file remains.
 		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
-			valid: true,
+			name:          "swap out file",
+			oldTempExists: false,
 		},
+	}
 
-		// Test with noBackupArchive set to true - should not create
-		// archive.
-		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
-			noBackupArchive: true,
-			valid:           true,
-		},
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tempTestDir := t.TempDir()
 
-		// Test with v set to false - should create
```

Comment:
We now enter this test, with the leftover state from the previous test, in which a file was left on the disk.